addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3835,11 +3835,22 @@ export class OdooEditor extends EventTarget {
             ev.stopPropagation();
             this.execCommand('strikeThrough');
         } else if (IS_KEYBOARD_EVENT_LEFT_ARROW(ev)) {
+            const selection = this.document.getSelection();
+            const codeElement = closestElement(selection.anchorNode, 'code');
+            if (
+                codeElement?.classList.contains('o_inline_code') &&
+                !selection.anchorOffset &&
+                !isZWS(codeElement?.previousSibling)
+            ) {
+                codeElement.before(document.createTextNode('\u200B'));
```

Comment:
The zws is originally only added when the code element is at the edge of its parent (`!codeElement.previousSibling || codeElement.previousSibling.nodeType !== Node.TEXT_NODE` is the check that is used). Make sure you don't add zws where they're not needed because that could quickly mess up selections.

Also, a little comment here would be welcome, so we can tell at a glance that what we're doing here is adding a zws before any inline code element if it didn't exist, and moving the selection next to it, and why.

Reply:
@Zynton i have added the comment and improve condition that if there is textNode before or after the codeElement the it won't add zws.